packages/tldraw/src/lib/shapes/image/ImageShapeUtil.tsx

```diff
@@ -59,16 +62,51 @@ export class ImageShapeUtil extends BaseBoxShapeUtil<TLImageShape> {
 		const isCropping = this.editor.getCroppingShapeId() === shape.id
 		const prefersReducedMotion = usePrefersReducedMotion()
 		const [staticFrameSrc, setStaticFrameSrc] = useState('')
+		const [loadedSrc, setLoadedSrc] = useState('')
+		const { useAssetHandler } = useEditorHooks()
+		const { handleAsset } = useAssetHandler()
 
 		const asset = shape.props.assetId ? this.editor.getAsset(shape.props.assetId) : undefined
 
 		const isSelected = shape.id === this.editor.getOnlySelectedShapeId()
 
+		// We debounce the zoom level to reduce the number of times we fetch a new image and,
+		// more importantly, to not cause zooming in and out to feel janky.
+		const [debouncedZoom, setDebouncedZoom] = useState(this.editor.getZoomLevel())
+		const zoomUpdater = useRef(debounce((zoom: number) => setDebouncedZoom(zoom), 500))
+		useValue('zoom level', () => zoomUpdater.current(this.editor.getZoomLevel()), [this.editor])
```

Comment:
This is a fun / clever way of writing a debounced reactor. attn @ds300

We did something similar in `useZoomCss` using an effect scheduler. I'm not sure whether it matters one way or the other, but seeing this kind of "debounced reactor" pattern is used twice then maybe it's something to add to the lib.